test/px-data-grid-test.js

```diff
@@ -37,12 +37,33 @@ function runTests() {
         grid.tableData = data;
       });
 
-      it('should properly populate _generatedColumns', function() {
+      it('should properly populate _generatedColumns', () => {
         expect(grid._generatedColumns.length).to.be.eql(3);
         expect(grid._generatedColumns[0].name).to.be.eql('first');
         expect(grid._generatedColumns[1].name).to.be.eql('last');
         expect(grid._generatedColumns[2].name).to.be.eql('email');
       });
+
+      it('should set _spinnerHiddenTimeout when assigning remoteDataProvider', () => {
+        grid.remoteDataProvider = (params, callback) => {};
+        expect(grid._spinnerHiddenTimeout).to.be.ok;
+      });
+
+      it('should hide spinner unless 500ms have passed', () => {
+        grid.remoteDataProvider = (params, callback) => {};
+        expect(grid.shadowRoot.querySelector('px-spinner').hasAttribute('hidden')).to.be.true;
+      });
+
+      it('should show spinner after 500ms when data is loading', () => {
+        const spy = sinon.spy(window, 'setTimeout');
```

Comment:
this test is hard to read and maintain, can it be simplified?

Reply:
Done.